src/v/archival/ntp_archiver_service.cc

```diff
@@ -158,14 +135,43 @@ ss::future<> ntp_archiver::outer_upload_loop() {
           })
           .handle_exception([this](std::exception_ptr e) {
               vlog(_rtclog.error, "upload loop error: {}", e);
-          })
-          .finally([this] {
-              vlog(_rtclog.debug, "upload loop stopped");
-              _probe.reset();
```

Comment:
Why did we remove this?

Reply:
The functional parts of the branch were no longer needed (resetting probe and updating _upload_loop_state), didn't seem worth keeping it around just to log.

However... logs are nice, I'm adding a commit that does this differently, logging at the top of the loop when it waits for leadership and/or proceeds, that will give us clear logs of how this code progresses.